### PR TITLE
[Feat] Mining Moles

### PR DESCRIPTION
**Pull request by Rofrtd**
# Description

This PR introduces the mining Moles.

T1 - Tunnel Mole
T2 - Rocky the Mole
T3 - Nugget

Screen Shot 2022-05-24 at 11 07 39 pm

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Screen Shot 2022-05-24 at 11 06 53 pm

Added tests for withdrawing cases and manually tested each tier.

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/goblins/bank/lib/bankUtil.test.ts [hunk 1]
```diff
@@ -302,7 +302,7 @@ describe("canWithdraw", () => {
     expect(enabled).toBeFalsy();
   });
 
-  it("prevent a user to withdraw kuebiko while they have seeds", () => {
+  it("prevents a user to withdraw kuebiko while they have seeds", () => {
```
Comment (brian-man) [inline src/features/goblins/bank/lib/bankUtil.test.ts:305]: "prevents" wasn't bad grammar but "to withdraw" should be "from withdrawing"

Comment (Rofrtd) [reply src/features/goblins/bank/lib/bankUtil.test.ts:305]: Yeah true, I can update this! Thanks for the feedback.

File: src/features/goblins/bank/lib/bankUtils.ts [hunk 2]
```diff
@@ -74,6 +75,27 @@ export function canWithdraw({ item, game }: CanWithdrawArgs) {
     return !cropIsPlanted({ item: "Parsnip", game });
   }
 
+  const stoneReady = Object.values(game?.stones).every((stone) =>
+    canMine(stone)
+  );
+  const ironReady = Object.values(game?.iron).every((iron) => canMine(iron));
```
Comment (brian-man) [inline src/features/goblins/bank/lib/bankUtils.ts:81]: Recommend: Move `ironReady` to below `Tunnel Mole` check and `goldReady` to below `Rocky the Mole` check for performance.

**Review by tiffanydys — COMMENTED** [review]
A few code recommendations 😄

**Review by c-py — APPROVED** [review]

**Review by adamhannigan — APPROVED** [review]
LGTM 🚀

**Review by tiffanydys — APPROVED** [review]

**Pull request merged**